test/trento/application/usecases/hosts_test.exs

```diff
@@ -38,4 +41,70 @@ defmodule Trento.HostsTest do
       refute deregistered_host.id in hosts_ids
     end
   end
+
+  describe "deregister_host" do
+    test "should deregister a host with unknown health" do
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{} ->
+          :ok
+        end
+      )
+
+      %{id: id} = insert(:host, heartbeat: Health.unknown())
+      assert :ok = Hosts.deregister_host(id)
+    end
+
+    test "should deregister a host with critical health" do
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{} ->
+          :ok
+        end
+      )
+
+      %{id: id} = insert(:host, heartbeat: Health.critical())
+      assert :ok = Hosts.deregister_host(id)
+    end
+
+    test "should not deregister a host with passing health" do
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{} ->
+          :error
+        end
+      )
+
+      %{id: id} = insert(:host, heartbeat: Health.passing())
+      assert :error = Hosts.deregister_host(id)
+    end
+
+    test "should return an error when the host does not exist" do
+      assert {:error, :host_not_found} = Hosts.deregister_host(UUID.uuid4())
+    end
+
+    test "should return an error if deregistration requested within the debounce period" do
+      last_heartbeat = DateTime.utc_now()
+
+      expect(
+        Trento.Support.DateService.Mock,
+        :utc_now,
+        fn -> last_heartbeat end
+      )
+
+      %{id: id} = insert(:host, heartbeat: Health.critical())
+      insert(:heartbeat, agent_id: id, timestamp: Trento.Support.DateService.Mock.utc_now())
+
+      expect(
+        Trento.Support.DateService.Mock,
+        :utc_now,
+        fn -> DateTime.add(last_heartbeat, 500, :millisecond) end
```

Comment:
Should we document that the default value of `Trento.Heartbeats.interval` is set to 6 seconds? Otherwise seeing this deregistration attempt within 500 ms of the last heartbeat seems a bit mysterious in my opinion.

Reply:
yes!

Reply:
This is already documented in the `config` file. Where or what do you plan to document?